src/main/java/org/jolokia/docker/maven/StartMojo.java

```diff
@@ -145,6 +147,39 @@ private void waitIfRequested(DockerAccess docker, ImageConfiguration imageConfig
             checkers.add(getLogWaitChecker(wait.getLog(), docker, containerId));
             logOut.add("on log out '" + wait.getLog() + "'");
         }
+        if (wait.getTcp() != null) {
+            WaitConfiguration.TcpConfiguration tcpConfig = wait.getTcp();
+            try {
+                Container container = docker.inspectContainer(containerId);
+                String host = tcpConfig.getHost();
+                List<Integer> ports = new ArrayList<>();
+
+                if (host == null) {
+                    // Host defaults to ${docker.host.address}.
+                    host = projectProperties.getProperty("docker.host.address");
+                }
+
+                if ("localhost".equals(host)) {
+                    host = container.getIPAddress();
+                    ports = tcpConfig.getPorts();
+                    log.info(String.format("%s: Waiting for ports %s directly on container with IP (%s).", imageConfig.getDescription(), ports, host));
+                } else {
+                    for (int port : tcpConfig.getPorts()) {
+                        Container.PortBinding binding = container.getPortBindings().get(port + "/tcp");
```

Comment:
binding can be null, too, when there configuration doesn't fit the port binding configuration. (leading to a NPE in the following line). This should be dealt with here (probably by throwing an exception because its an configuration mismatch).